### ISSUE TITLE
airspec: `shouldBe` matcher fails to detect null and undefined parameters in Scala 3 + Scala.js

### ISSUE THREAD

**Issue by xerial** [wvlet/airframe#2426]
The cause is Scala 3 compiler failed to generate val fields of null or undefined for js.Object https://github.com/lampepfl/dotty/issues/14168

For example, this test fails: 
```scala
    val c = new js.Object() {
      val a     = 1
      val extra = js.undefined
    }
    val d = new js.Object() {
      val a = 1
    }
    c shouldNotBe d  // This fails because the extra field (undefined) is not generated by dotty
```

```
$ cd airframe/airspec
$ DOTTY=true ../sbt
> airspecJS/testOnly jsexamples.JSMatcherTest
```

**Comment by xerial** [comment]
It's fixed in https://github.com/lampepfl/dotty/pull/16187. Need to wait for the next Scala 3 release

**Comment by xerial** [comment]
This will be fixed in Scala 3.2.2 https://github.com/lampepfl/dotty/releases/tag/3.2.2-RC1
